tasks/update.yml

```diff
@@ -0,0 +1,31 @@
+---
+# Creates a folder /opt/mattermost/{{ mattermost_version }} with the desired version
+
+- set_fact:
+    mattermost_archive_name: mattermost-team-{{ mattermost_version }}
+  when: not mattermost_enterprise
+
+- set_fact:
+    mattermost_archive_name: mattermost-{{ mattermost_version }}
+  when: mattermost_enterprise
+
+- name: 'Download binary from https://releases.mattermost.com'
+  get_url:
+    url: 'https://releases.mattermost.com/{{ mattermost_version }}/{{ mattermost_archive_name }}-linux-amd64.tar.gz'
+    dest: "{{ global_cache_dir | mandatory }}/"
+  become: false
+  delegate_to: localhost
+
+- name: Unpack Mattermost archive
+  unarchive:
+    src: "{{ global_cache_dir }}/{{ mattermost_archive_name }}-linux-amd64.tar.gz"
+    dest: /tmp
```

Comment:
Why not unpack the archive directly to `/opt/mattermost-versions/{{ mattermost_version }}`?

Reply:
Because `install.yml` assumes that every folder inside `/opt/mattermost-versions` contains a fully and successfully installed version.